pkg/systemd/overview.jsx

```diff
@@ -0,0 +1,159 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import 'polyfills.js';
+import cockpit from "cockpit";
+import $ from "jquery";
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {
+    Page, PageSection, PageSectionVariants,
+    Gallery, Button,
+    Dropdown, DropdownItem, KebabToggle,
+} from '@patternfly/react-core';
+
+import { shutdown, shutdown_modal_setup } from "./shutdown.js";
+
+import { SystemInfomationCard } from './overview-cards/systemInformationCard.jsx';
+import { ConfigurationCard } from './overview-cards/configurationCard.jsx';
+import { HealthCard } from './overview-cards/healthCard.jsx';
+import { MotdCard } from './overview-cards/motdCard.jsx';
+import { UsageCard } from './overview-cards/usageCard.jsx';
+import { ServerTime } from './overview-cards/serverTime.js';
+
+import "form-layout.less";
+
+const _ = cockpit.gettext;
+var permission = cockpit.permission({ admin: true });
+$(permission).on("changed", update_shutdown_privileged);
+
+function update_shutdown_privileged() {
+    $(".shutdown-privileged").update_privileged(
+        permission, cockpit.format(
+            _("The user <b>$0</b> is not permitted to shutdown or restart this server"),
+            permission.user ? permission.user.name : '')
+    );
+}
+
+class OverviewPage extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            actionKebabIsOpen: false
+        };
+        this.onKebabToggle = actionKebabIsOpen => {
+            this.setState({
+                actionKebabIsOpen
+            });
+        };
```

Comment:
I know it's just a whim of mine, but can these simple handlers please just use one line? And drop the unnecessary { } wrappers and newlines?

    this.onKebabToggle = actionKebabIsOpen => this.setState({ actionKebabIsOpen });

is  IMHO not harder to read, and avoids the inflation of lines with interpunction.

If you want to go more fancy, this kind of "simple state" would also be an excellent usage of hooks (see commit 286bfd790fa75a6901c4d54319f05aa4336fe9ff)

Reply:
Fixed.